src/Elmish.WPF/BindingVmHelpers.fs

```diff
@@ -214,6 +214,100 @@ and VmBinding<'model, 'msg, 't> =
+    | SubModel b -> SubModel {
+        SubModelData = {
+          GetModel = b.SubModelData.GetModel
+          CreateViewModel = b.SubModelData.CreateViewModel >> fOut
+          UpdateViewModel = (fun (vm,m) -> b.SubModelData.UpdateViewModel (fIn vm, m))
+          ToMsg = b.SubModelData.ToMsg }
+        Dispatch = b.Dispatch
+        GetVm = b.GetVm >> ValueOption.map fOut
+        SetVm = ValueOption.map fIn >> b.SetVm
+        GetCurrentModel = b.GetCurrentModel }
+    | SubModelWin b -> SubModelWin {
+        SubModelWinData = {
+          GetState = b.SubModelWinData.GetState
+          CreateViewModel = b.SubModelWinData.CreateViewModel >> fOut
+          UpdateViewModel = (fun (vm,m) -> b.SubModelWinData.UpdateViewModel (fIn vm, m))
+          ToMsg = b.SubModelWinData.ToMsg
+          GetWindow = b.SubModelWinData.GetWindow
+          IsModal = b.SubModelWinData.IsModal
+          OnCloseRequested = b.SubModelWinData.OnCloseRequested }
+        Dispatch = b.Dispatch
+        WinRef = b.WinRef
+        PreventClose = b.PreventClose
+        GetVmWinState = b.GetVmWinState >> WindowState.map fOut
+        SetVmWinState = WindowState.map fIn >> b.SetVmWinState
+        GetCurrentModel = b.GetCurrentModel }
+    | SubModelSeqUnkeyed b -> SubModelSeqUnkeyed {
+        SubModelSeqUnkeyedData = {
+          GetModels = b.SubModelSeqUnkeyedData.GetModels
+          CreateViewModel = b.SubModelSeqUnkeyedData.CreateViewModel
+          CreateCollection = b.SubModelSeqUnkeyedData.CreateCollection >> CollectionTarget.mapCollection fOut
+          UpdateViewModel = b.SubModelSeqUnkeyedData.UpdateViewModel
+          ToMsg = b.SubModelSeqUnkeyedData.ToMsg }
+        Dispatch = b.Dispatch
+        Vms = b.Vms |> CollectionTarget.mapCollection fOut
+        GetCurrentModel = b.GetCurrentModel }
+    | SubModelSeqKeyed b -> SubModelSeqKeyed {
+        SubModelSeqKeyedData = { 
+          GetSubModels = b.SubModelSeqKeyedData.GetSubModels
+          CreateViewModel = b.SubModelSeqKeyedData.CreateViewModel
+          CreateCollection = b.SubModelSeqKeyedData.CreateCollection >> CollectionTarget.mapCollection fOut
+          UpdateViewModel = b.SubModelSeqKeyedData.UpdateViewModel
+          ToMsg = b.SubModelSeqKeyedData.ToMsg
+          BmToId = b.SubModelSeqKeyedData.BmToId
+          VmToId = b.SubModelSeqKeyedData.VmToId }
+        Dispatch = b.Dispatch
+        Vms = b.Vms |> CollectionTarget.mapCollection fOut
+        GetCurrentModel = b.GetCurrentModel }
+    | SubModelSelectedItem b -> SubModelSelectedItem {
+        Get = b.Get
+        Set = b.Set
+        SubModelSeqBindingName = b.SubModelSeqBindingName
+        SelectedItemBinding = {
+          VmToId = fIn >> b.SelectedItemBinding.VmToId
+          FromId = b.SelectedItemBinding.FromId >> Option.map fOut } }
+
+  let rec private recursivecase<'model, 'msg, 'a, 'b> (fOut: 'a -> 'b) (fIn: 'b -> 'a) (data: VmBinding<'model, 'msg, 'a>) : VmBinding<'model, 'msg, 'b> =
```

Comment:
Did you mean to capitalize as `recursiveCase`? Doesn't necessarily need to be fixed for this PR.

Reply:
Go ahead and push a change

Reply:
Done.